### PR TITLE
README with enhanced install instructions for protobuf

### PR DESCRIPTION
**Pull request by PhRosenberger**
﻿#### Reference to a related issue in the repository
Relates to
- #592
- https://github.com/OpenSimulationInterface/osi-sensor-model-packaging/issues/87

#### Add a description
We need better install instructions for static / dynamic linking, so this PR should enhance the README for this.
More explanation should then follow in the OSI documentation, I guess?

#### Take this checklist as orientation for yourself, if this PR is ready for the Change Control Board:
- [ x] My suggestion follows the [style and contributors guidelines](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/contributing/start_contributing.html).
- [x] I have taken care about the [message documentation](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/contributing/commenting_messages.html) and the [fields and enums documentation](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/contributing/commenting_fields_enums.html).
- [x] I have done the [DCO signoff](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html#developer-certification-of-origin-dco).
- [x] My changes generate no errors when passing CI tests. 
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

### CODE REVIEW

**Comment by jdsika** [comment]
Use placeholder for version

**Comment by PhRosenberger** [comment]
@pmai, could you please provide the further details on proper linker flags in addition to the instructions in the new file installing_prerequisites.adoc?

File: doc/setup/installing_prerequisites.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Static linking is the recommended way to package OSI FMUs for sharing with others to be integrated in a co-simulation: They should have minimal external dependencies, in order to ensure maximum portability.
+Fo other use cases like first trials to get started with OSI, dynamic linking could be an option, as well.
+
+== Windows
+
+**Dynamic Linking (NOT RECOMMENDED)**
+
+Since on windows symbols are not exported per default it can be kind of annoying to deal with this during _protobuf_ header generation (see for example  https://groups.google.com/g/protobuf/c/PDR1bqRazts).
+So basically thats one more reason to use static linking.
```
Comment (philipwindecker) [inline doc/setup/installing_prerequisites.adoc]: Would rephrase as follows:

```suggestion
That is an important reason to use static linking.
```

File: doc/setup/installing_prerequisites.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+* You have installed _cmake_ (version 3.10.2 or higher).
+* You must have super user privileges.
+
+Static linking is the recommended way to package OSI FMUs for sharing with others to be integrated in a co-simulation: They should have minimal external dependencies, in order to ensure maximum portability.
+Fo other use cases like first trials to get started with OSI, dynamic linking could be an option, as well.
+
+== Windows
+
+**Dynamic Linking (NOT RECOMMENDED)**
```
Comment (philipwindecker) [inline doc/setup/installing_prerequisites.adoc:18]: I would recommend using description lists instead of manually formatting as bold:

```suggestion
Dynamic Linking (NOT RECOMMENDED)::
```

Comment (philipwindecker) [reply doc/setup/installing_prerequisites.adoc:18]: This goes for all the following examples as well.
However, the empty line between this and its description should also be removed then.

**Comment by PhRosenberger** [comment]
@philipwindecker Thanks a lot for your support with documentation references and your review!

**Review by philipwindecker — CHANGES_REQUESTED** [review]
Review done. I would recommend using description lists instead of bold formatting because they provide more consistent styling. See https://docs.asciidoctor.org/asciidoc/latest/lists/description/
https://docs.asciidoctor.org/asciidoc/latest/lists/horizontal/

Alternatively, using subsections (level 2 i.e. `===`) would help with automatically creating anchors and entries in the toc.

File: doc/setup/installing_prerequisites.adoc (hunk 2 above)
Comment (philipwindecker) [reply doc/setup/installing_prerequisites.adoc:18]: Ideally: yes. However, this could be done in a separate cleanup issue then.

It is not a required step ofc.  It's just that I personally feel that using the included functionality of sections or description lists for this type of section declarations is better.

However, if you would like to stick to the current styling (i.e. not numbered, same font size, and no tabulation), it might even be the best approach.

Comment (PhRosenberger) [reply doc/setup/installing_prerequisites.adoc:18]: Ok, so let's put it on the agenda for the next CCB and discuss if this should be opened as a separate issue there.

Comment (PhRosenberger) [reply doc/setup/installing_prerequisites.adoc]: As decided in the CCB, I have changed the PR now, using description lists instead of manually formatting as bold in doc/setup.

**Comment by pmai** [comment]
CCB 2023-05-22: Merge as-is, conflicts to be resolved @pmai

**Review by pmai — APPROVED** [review]
